ckanext/canada/view.py

```diff
@@ -579,6 +625,64 @@ def ckanadmin_publish_datasets():
     return h.redirect_to('canada.ckanadmin_publish')
 
 
+@canada_views.route('/ckan-admin/job-queue', methods=['GET'])
+def ckanadmin_job_queue():
+    """
+    Lists all of the queued and running jobs.
+    """
+    try:
+        jobs = get_action('job_list')({}, {})
+    except NotAuthorized:
+        abort(403, _('Not authorized to see this page'))
+
+    warning = False
+    if jobs and datetime.strptime(jobs[0]['created'], '%Y-%m-%dT%H:%M:%S') < (datetime.now() - timedelta(minutes=18)):
+        warning = True
+
+    job_list = []
+    for job in jobs[:MAX_JOB_QUEUE_LIST_SIZE]:
```

Comment:
MAX_JOB_QUEUE_LIST_SIZE should be a job_list `limit` parameter instead to save wasted work getting jobs that we discard.

Reply:
@wardi here is a PR to our fork/branch to add support for offset & limit to the `job_list` action method. https://github.com/open-data/ckan/pull/165

I noted in the PR about an issue if we want to submit that upstream.